ice40/arch.cc

```diff
@@ -1231,6 +1242,21 @@ void Arch::assignCellInfo(CellInfo *cell)
             cell->lcInfo.inputCount++;
         if (cell->getPort(id_I3))
             cell->lcInfo.inputCount++;
+        // Find don't care LUT inputs to mask for timing analysis
+        cell->lcInfo.lutInputMask = 0x0;
+        unsigned init = int_or_default(cell->params, id_LUT_INIT);
+        for (unsigned k = 0; k < 4; k++) {
+            bool care = false;
+            for (unsigned i = 0; i < 16; i++) {
+                // If toggling the LUT input makes a difference it's not a don't care
+                if (((init >> i) & 0x1U) != ((init >> (i ^ (1U << k))) & 0x1U)) {
+                    care = true;
+                    break;
+                }
```

Comment:
Hmm. How about 
```suggestion
                // If toggling the LUT input makes a difference it's not a don't care
                care = ((init >> i) & 0x1U) != ((init >> (i ^ (1U << k))) & 0x1U));
                if (care)
                    break;
```

Reply:
Actually, if you have an `if (care)` condition, you might as well also insert the contents of the `if (care)` block below this loop.

Reply:
not sure what you mean by below this loop? and I must admit the original is more in line with my personal style, although the suggestion is pretty reasonable too.

Reply:
oh I see, I can get rid of `care` altogether, that makes sense